extmod/moduzlib.c

```diff
@@ -206,9 +207,62 @@ STATIC mp_obj_t mod_uzlib_decompress(size_t n_args, const mp_obj_t *args) {
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(mod_uzlib_decompress_obj, 1, 3, mod_uzlib_decompress);
 
 #if !MICROPY_ENABLE_DYNRUNTIME
+STATIC mp_obj_t mod_uzlib_gzip(size_t n_args, const mp_obj_t *args) {
+    mp_obj_t data = args[0];
+    mp_buffer_info_t bufinfo;
+    mp_get_buffer_raise(data, &bufinfo, MP_BUFFER_READ);
+    int len = bufinfo.len;
+
+    struct uzlib_comp *comp = m_new_obj(struct uzlib_comp);
+    memset(comp, 0, sizeof(*comp));
+
+    comp->dict_size = 32768;
+    comp->hash_bits = 12;
+    size_t hash_size = sizeof(uzlib_hash_entry_t) * (1 << comp->hash_bits);
+    comp->hash_table = gc_alloc(hash_size, false);
+    memset(comp->hash_table, 0, hash_size);
+
+    zlib_start_block(&comp->out);
+    uzlib_compress(comp, bufinfo.buf, len);
+    zlib_finish_block(&comp->out);
+
+    printf("compressed from %u to %u raw bytes\n", len, comp->out.outlen);
+
+    mp_uint_t dest_buf_size = (comp->out.outlen + 6 + (3*sizeof(int)));
+    byte *dest_buf = m_new(byte, dest_buf_size);
+
+    int i = 0;
+    dest_buf[i++] = 0x1f;
+    dest_buf[i++] = 0x8b;
+    dest_buf[i++] = 0x08;
+    dest_buf[i++] = 0x00; // FLG
+    int mtime = 0;
+    memcpy(&dest_buf[i], &mtime, sizeof(mtime));
+    i += sizeof(mtime);
```

Comment:
According to [Wikipedia](https://en.wikipedia.org/wiki/Gzip), `mtime` must be 4 bytes. Use `uint32_t` here?